ops/charm.py

```diff
@@ -1488,6 +1562,97 @@ def __init__(self, name: str, raw: Dict[str, Any]):
         self.type = raw['type']
 
 
+class DeviceType(enum.Enum):
+    """Types of device that may be requested."""
+    GPU = 'gpu'
+    NVIDIA_GPU = 'nvidia.com/gpu'
+    AMD_GPU = 'amd.com/gpu'
```

Comment:
I wonder if we should make this just a str rather than an enum with defined values? These seem pretty specific, and we don't necessarily want to have to update this if stuff is added to the list. We could do the `str | DeviceType` trick as I've done elsewhere, but that's kind of annoying.

Reply:
Sure. I did cringe a bit at it having nvidia & amd in there rather than nice generic things like GPU, NFC, BLE etc :joy:.